src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -308,5 +248,22 @@ internal static void ValidateCachePath(string path)
                 throw new ArgumentException($"The directory with path {directoryPath} does not exist.");
             }
         }
+
+        internal static void ValidateFileCacheExpiration(TimeSpan expiration)
+        {
+            if (expiration <= TimeSpan.Zero)
+            {
+                throw new ArgumentException($"Please provide a valid {nameof(OfflineFileCacheOptions)}.{nameof(OfflineFileCacheOptions.FileCacheExpiration)}.");
+            }
+
+            try
+            {
+                _ = DateTimeOffset.UtcNow + expiration;

```

Comment:
Can this be rewritten without exception ? `DateTimeOffset.MaxValue - expiration > DatetimeOffset.UtcNow

Reply:
I tried that too. If `expiration` is `TimeSpan.MaxValue`, `DateTimeOffset.MaxValue - expiration` also threw `ArgumentOutOfRangeException`

Reply:
This validation is interesting. We use expiration at later points during runtime, at which the addition here may overflow. We're validating something who's validity depends on when it's validated. It may require being more restrictive on upper limit to be reasonable.

Reply:
I agree that this addition could still overflow at the time when data expiration is actually calculated. But we can't really have an arbitrary upper limit here. DataProtection docs suggest that the expiration should be no longer than a few weeks/one month. But they also don't have any enforcement for the upper limit. They just bubble up `ArgumentOutOfRangeException` if expiration is too big.

The only reason I want to keep this check here is because there might be some users who might try to persist their cache file forever and will therefore pass `TimeSpan.MaxValue` as their file expiration. For such users, this is an error that can be thrown much earlier during application startup instead of failing at the end of startup.

Reply:
Moved this logic to `Export` because we need to calculate the expiration time anyway.